src/overmapbuffer.cpp

```diff
@@ -598,9 +598,17 @@ void overmapbuffer::remove_vehicle( const vehicle *veh )
     om_loc.om->vehicles.erase( veh->om_id );
 }
 
-void overmapbuffer::add_vehicle( vehicle *veh )
+void overmapbuffer::add_vehicle( vehicle *veh, const point &abs_override )
```

Comment:
I'm strongly against this change. Set `vehicle::sm_pos` prior to calling the function if you must (the way it works for now), but "overriding" it introduces a discrepancy in the data (`vehicle::sm_pos` still stays the same), not mentioning worsening the interface.

Reply:
I tried that way previously, but Im afraid to admit, I couldnt get it to work, hence the ugly brute force override. The coordinate conversions are confusing me, and I dont know what to set the veh->sm_pos as, I tried omt_to_sm_copy( goal ) , but that dosnt work.